substrate/frame/support/src/lib.rs

```diff
@@ -2262,6 +2262,221 @@ pub mod pallet_macros {
+	///
+	/// ### Example:
+	/// 
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	///     # use frame_support::pallet_prelude::*;
+	///     # #[pallet::config]
+	///     # pub trait Config: frame_system::Config {}
+	///     # #[pallet::pallet]
+	///     # pub struct Pallet<T>(_);
+	/// 	#[pallet::storage]
+	/// 	type Foo<T> = StorageValue<_, u32>;
+	///
+	/// 	#[pallet::storage]
+	/// 	type Bar<T> = StorageMap<_, Blake2_128Concat, u32, u32>;
+	///
+	/// 	#[pallet::storage]
+	/// 	type Baz<T> = CountedStorageMap<_, Blake2_128Concat, u32, u32>;
+	/// }
+	/// ```
+	/// 
+	/// ## Related Macros
+	/// 
+	/// The following macros can be used in conjunction with the storage macro:
+	///
+	/// * [`macro@getter`]: Creates a custom getter function.
+	/// * [`macro@storage_prefix`]: Overrides the default prefix of the storage item.
+	/// * [`macro@unbounded`]: Declares the storage item as unbounded.
+	///
+	/// ## Common Details to All Storage Types.
+	///
+	/// The following details are relevant to all of the aforementioned storage types.
+	///
+	/// ### Syntax
+	///
+	/// Two general syntaxes are supported, as demonstrated below:
+	///
+	/// 1. Named generics, e.g., `type Foo<T> = StorageValue<Value = u32>`.
+	/// 2. Unnamed generics, e.g., `type Foo<T> = StorageValue<_, u32>`.
+	///
+	/// In both instances, declaring `<T>` is mandatory. While it can optionally
+	/// be written as `<T: Config>`, in the generated code, it is always `<T: Config>`.
+	///
+	/// #### Example:
+	/// 
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	///     # use frame_support::pallet_prelude::*;
+	///     # #[pallet::config]
+	///     # pub trait Config: frame_system::Config {}
+	///     # #[pallet::pallet]
+	///     # pub struct Pallet<T>(_);
+	/// 	/// Unnamed syntax, without bounding `T`.
+	///     #[pallet::storage]
+	///     pub type Foo<T> = StorageValue<_, u32>;
+	///
+	/// 	/// Unnamed syntax, with bounding `T`.
+	///     #[pallet::storage]
+	///     pub type Bar<T: Config> = StorageValue<_, u32>;
+	///
+	/// 	/// Named syntax.
+	///     #[pallet::storage]
+	///     pub type Baz<T> = StorageMap<_, Hasher = Blake2_128Concat, Key = u32, Value = u32>;
+	/// }
+	/// ```
+	///
+	/// ### Query Type
+	///
+	/// Every storage type mentioned above has a generic type called `QueryKind` that determines its "query" type. 
+	/// This refers to the kind of value returned when querying the storage, for instance, through a `::get()` method. 
+	///
+	/// There are three types of queries:
+	///
+	/// 1. [`OptionQuery`](frame_support::storage::types::OptionQuery): The default query type. It returns `Some(V)` if the value is present, or 
+	/// `None` if it isn't, where `V` is the value type.
+	/// 2. [`ValueQuery`](frame_support::storage::types::ValueQuery): Returns the value itself if present; otherwise, it returns 
+	/// `Default::default()`. This behavior can be adjusted with the `OnEmpty` generic parameter, which defaults to `OnEmpty = GetDefault`.
```

Comment:
Would like to see somewhere (probably in the documentation of `ValueQuery`) an example of a storage value that is u32 who's default getter returns 42 if the value is not present.